public/pages/Snapshots/components/RestoreActivitiesPanel/RestoreActivitiesPanel.tsx

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiInMemoryTable, EuiSpacer, EuiLink, EuiFlyout, EuiButton } from "@elastic/eui";
+import _ from "lodash";
+import React, { useEffect, useContext, useState } from "react";
+import { SnapshotManagementService } from "../../../../services";
+import { CoreServicesContext } from "../../../../components/core_services";
+import { getErrorMessage } from "../../../../utils/helpers";
+import { GetIndexRecoveryResponse, CatSnapshotIndex } from "../../../../../server/models/interfaces";
+import { BREADCRUMBS } from "../../../../utils/constants";
+import { ContentPanel } from "../../../../components/ContentPanel";
+
+interface RestoreActivitiesPanelProps {
+  snapshotManagementService: SnapshotManagementService;
+  snapshotId: string;
+  repository: string;
+}
+
+export const RestoreActivitiesPanel = ({ snapshotManagementService, snapshotId }: RestoreActivitiesPanelProps) => {
+  const context = useContext(CoreServicesContext);
+  const [startTime, setStartTime] = useState("");
+  const [stopTime, setStopTime] = useState("");
+  const [stage, setStage] = useState("");
+  const [indices, setIndices] = useState([{}]);
+  const [flyout, setFlyout] = useState(false);
+
+  useEffect(() => {
+    context!.chrome.setBreadcrumbs([BREADCRUMBS.SNAPSHOT_MANAGEMENT, BREADCRUMBS.SNAPSHOTS, BREADCRUMBS.SNAPSHOT_RESTORE]);
```

Comment:
Is there a way to avoid this? I know we might not have the required types in place, but can we find what it would take and create an issue for that?

Reply:
Are you referring to the ! ?  looks like it can be replaced with '?'.  Will do if that is the issue.

Reply:
i did it anyway.

Reply:
yeah I am referring to the `!`, weird it didn't highlight it. Yeah we can use `?` but that would still be like a bandaid if we know for sure that the context is bound to be present.
Okay for now, but let's investigate if we can update the types to reflect the correct behavior

Reply:
not sure how else to resolve - says const context: CoreStart | null

Reply:
it's in the OSD domain, so I guess a ticket would be the way, but it may be needed for other situations the way it is.  What do you think?

Reply:
yeah, a ticket would make sense. Let me look into it